sklearn/ensemble/gradient_boosting.py

```diff
@@ -965,8 +964,41 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # fit initial model - FIXME make sample_weight optional
             self.init_.fit(X, y, sample_weight)
```

Comment:
Actually this is weird: the tests checks with pre-trained models and this does not assume that `init_` has been trained. It seems inconsistent.

Also if we assume `self.init_` to be pre-trained we should set `n_targets = self.init_.classes_.shape[0]` for a classifier.